extension/src/extension.ts

```diff
@@ -59,26 +59,14 @@ export async function activate(context: ExtensionContext) {
     testResourceManager.onDidChangeTestStorage((e) => {
         testExplorer.refresh();
     });
-
-    workspace.onDidChangeTextDocument((event) => {
-        const uri = event.document.uri;
-        testResourceManager.setDirty(uri);
-        // onDidChange.fire();
-    });
-
-    workspace.onDidCloseTextDocument((document) => {
-        const uri = document.uri;
-        // if a test file is removed, should update test storage.
-        if (!fs.existsSync(uri.fsPath)) {
-            testResourceManager.removeTests(uri);
-        }
-    });
-
-    workspace.onDidSaveTextDocument((document) => {
-        const uri = document.uri;
+    const watcher = workspace.createFileSystemWatcher('**/*.{[jJ][aA][vV][aA]}');
```

Comment:
Looks like we can push the `watcher` into `context.subscription`?

https://github.com/Microsoft/vscode/blob/80d6fa0b8eb5f4c0959b9e2925ec7fd28692880f/src/vs/vscode.d.ts#L1490
```javascript
export interface FileSystemWatcher extends Disposable
```

Reply:
yes, we should!